Fix: Mobile MVP UI and feature fixes

caybro opened the pull request:
### What does the PR do

This PR polishes the Mobile MVP experience across onboarding, account creation, navbar context menus, wallet settings, and Android file access. It also includes small Storybook and i18n housekeeping updates.

Here is a revised description of the pull request, with references to the issues fixed by each commit:

- **AppMain: Restore the forgotten Node section** Fixes #19050
- **Set `production` variant as default for mobile builds**: Updated the mobile Makefile to set the `production` variant by default, aligning it with the desktop variant. This change impacts logging and enables development features in the app ([Commit](https://github.com/status-im/status-desktop/commit/41311414b7230bd45be271031eb0d115cd08dc55)).
- **Use `nproc` in MAKEFLAGS for CI**: Replaced hardcoded usage of 4 CPU cores with `nproc` to dynamically determine available cores ([Commit](https://github.com/status-im/status-desktop/commit/7696408ddf5f0827f3fd70992c2e1e0a12d09b19)).
- **Fix file access on Android for tablets**: Adjusted permissions and file-saving paths to improve file access support on Android tablets. Fixes [#18076](https://github.com/status-im/status-desktop/issues/18076) and Fixes [#19053](https://github.com/status-im/status-desktop/issues/19053) ([Commit](https://github.com/status-im/status-desktop/commit/0deb35448fda6f5d0195798b9ddee21650812dfc)).
- **Adjust Beta tag padding in Wallet settings**: Fixed padding issues for the Beta tag in Wallet settings. Iterates on [#19068](https://github.com/status-im/status-desktop/issues/19068) ([Commit](https://github.com/status-im/status-desktop/commit/132da8e473b685b92a55811d80e7eea7f99891b9)).
- **Ensure Send button appears for Payments in Chat Input**: Updated StatusChatInput to display the Send button for Payments. Fixes [#19049](https://github.com/status-im/status-desktop/issues/19049) ([Commit](https://github.com/status-im/status-desktop/commit/5a8537065953daac9134f9dbf274271d4e7c752e)).
- **Fix layout issues in Advanced Settings view**: Corrected layout issues and adjusted the StatusSettingsLineButton component for mobile. Fixes [#19072](https://github.com/status-im/status-desktop/issues/19072) ([Commit](https://github.com/status-im/status-desktop/commit/116da8f1cd3b01a56a27ee074f8eb3af4fcda117)).
- **Simplify popup menu handling in Navbar**: Refactored context menu code to simplify handling of the Mute submenu and other interactions. Fixes [#19070](https://github.com/status-im/status-desktop/issues/19070) ([Commit](https://github.com/status-im/status-desktop/commit/e4293fdd6d5fd27b4cb2edaf3551f6ef0f590fe6)).
- **Fix token list display in Wallet Settings**: Resolved issues where the token list showed zero tokens and removed unused signals. Fixes [#19071](https://github.com/status-im/status-desktop/issues/19071) ([Commit](https://github.com/status-im/status-desktop/commit/6c506ce9de48a5d593c9bd0500e45874ee75d472)).
- **Fix tooltip rendering in StatusToolTip**: Adjusted size calculations and removed hardcoded metrics for better tooltip rendering. Fixes [#19059](https://github.com/status-im/status-desktop/issues/19059) ([Commit](https://github.com/status-im/status-desktop/commit/1485fc51b2047cd8ba506c8beeba96ffdcd94d6d)).
- **Reduce memory usage in Home Screen**: Replaced Glow effect with RectangularShadow for a faster and less resource-intensive rendering. Fixes [#19086](https://github.com/status-im/status-desktop/issues/19086) ([Commit](https://github.com/status-im/status-desktop/commit/50797925fea5a90f3bbb248e0a7006d7da877d6f)).
- **Fix layout for Login Screen profile selector**: Updated the LoginScreen to avoid using a bottom sheet for profile selection ([Commit](https://github.com/status-im/status-desktop/commit/6a269c5c75181b69ccaccefd0e32078303ceb2f5)).
- **Add bottom margin for Onboarding SafeArea**: Resolved issues by adding a bottom margin to the Onboarding SafeArea. Fixes [#19074](https://github.com/status-im/status-desktop/issues/19074) ([Commit](https://github.com/status-im/status-desktop/commit/9765fb6492a4d9c247bcedca93998b2b0b7c825d)).
- **Ignore generated files for i18n and update TS**: Excluded TS files within build directories ([Commit](https://github.com/status-im/status-desktop/commit/2f211a1f45de0fe567f7f2cefc40ab00a549fbe2)).
- **Update macOS minimum version for Storybook**: Raised the minimum macOS version to 14.0 ([Commit](https://github.com/status-im/status-desktop/commit/4b465c5793f422ba1e701d80908cf6814b819cb1)).

This PR introduces a series of fixes and improvements across different features, optimizing performance, usability, and developer experience. Each commit addresses specific issues or introduces enhancements as detailed above.

- Needs https://github.com/status-im/dotherside/pull/100
- Needs https://github.com/status-im/nimqml/pull/68

### Affected areas

App

### Architecture compliance

- [x] I am familiar with the [application architecture](/docs/architecture.md) and agreed good practices.
My PR is consistent with this document: [QML Architecture Guidelines](/guidelines/QML_ARCHITECTURE_GUIDE.md)

### Screencapture of the functionality

Profile image saved:
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/79b6b9dd-23dd-48eb-aca5-a8ee857f19ce" />

Sending an image in chat:
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/82b1b0d2-286e-42c9-b673-907485bd49b1" />

About (`production` logo):
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/ecdfe852-b6fc-491c-af20-6f52f1fe3fdf" />

Advanced section (fixed layout, disabled Browser/Debug parts):
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/4f1bc828-2a66-4ff7-a6d5-9ed202c50880" />

Home screen (new hover/glow effect):
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/b72ffdd9-b3a1-4b62-8fc3-e62eaa81bf3a" />

Token list (fixed popup header; icon/name/count/timestamp):
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/e6512291-f0f4-4554-9e8c-bf26bc20fb61" />

Node section restored:
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/a4476237-26c4-45ae-8656-8f93ead03a74" />

Cascading submenu fixed:
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/36ed2dbb-ae6b-40ee-a236-d24a147ccbfa" />

Send button enabled for transactions:
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/9935b682-7273-4b0e-bcc0-9830c2a59c66" />

Wallet beta tag alignment fixed:
<img width="576" height="1280" alt="image" src="https://github.com/user-attachments/assets/8c009fd2-c4bf-4d15-96c7-af339119f8ef" />

### Impact on end user

Stuff works

### How to test

TBD

### Risk 

low

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 0deb3544 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-ui/job/PR-19095/1/) | 2025-10-20 21:44:00 | ~2 min | `tests/ui` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-ui/job/PR-19095/1/consoleText) |

caybro commented on mobile/android/qt6/AndroidManifest.xml:
```diff
@@ -13,19 +13,19 @@
+    <uses-permission android:name="android.permission.USE_BIOMETRIC" />
+
+    <!-- features -->
+    <uses-feature android:name="android.hardware.camera" android:required="false" />
```
CHECKME

caybro commented on ci/Jenkinsfile.tests-ui:
```diff
@@ -70,8 +70,10 @@ pipeline {
       steps {
-        script {
-          checkTranslations()
+        catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
```
I made the TS file check optional; there's a Qt bug which doesn't catch `qsTr()` calls inside JS backticks

CC @siddarthkay @markoburcul

caybro replied:
Like here: https://github.com/status-im/status-desktop/blob/59bb23eef4ffec42601a439cdc6ddfd0433d1a82/ui/app/AppLayouts/Onboarding/pages/KeycardEnterPinPage.qml#L126-L129

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -1234,6 +1234,11 @@ Item {
                         value: Constants.appSection.browser
                         enabled: d.isBrowserEnabled
                     }
+                    ValueFilter {
```
Node section restored

micieslak commented on ui/app/AppLayouts/Profile/views/wallet/MainView.qml:
```diff
@@ -150,7 +150,7 @@ Column {
             anchors.verticalCenter: parent.verticalCenter
             anchors.left: parent.left
-            anchors.leftMargin: accountOrderItem.statusListItemTitle.width + Theme.bigPadding
+            anchors.leftMargin: accountOrderItem.statusListItemTitle.width + parent.leftPadding + Theme.bigPadding
```
Looks a bit hacky to be honest. On narrow screen it looks ok but on wider one the space seems to be too wide. Something with this calculation must be wrong if bigpadding is not enough and stuff was overlapping. It should not overlap even with 0 margin.

caybro commented:
> Nice work! Although I have a few findings while testing
> 
> > Ensure Send button appears for Payments in Chat Input: Updated StatusChatInput to display the Send button for Payments. Fixes #19049 ([Commit](https://github.com/status-im/status-desktop/commit/5a8537065953daac9134f9dbf274271d4e7c752e)).
> 
> The send button will show up when adding a payment request. But it does nothing. Cannot send the message

Yeah I mentioned that in the comments; there's a separate bug somewhere in NIM/status-go in that it won't let the user send the message w/o text

> 
> > Simplify popup menu handling in Navbar: Refactored context menu code to simplify handling of the Mute submenu and other interactions. Fixes #19070 ([Commit](https://github.com/status-im/status-desktop/commit/e4293fdd6d5fd27b4cb2edaf3551f6ef0f590fe6)).
> 
> I can use the submenu. But once the submenu is clicked there will be some menu leftovers on the screen
> 
> See the video

I'm not getting that... and frankly looks like a Qt redraw bug :(

>  Screen_Recording_20251027_092236_Status.mp4 
> > Fix file access on Android for tablets: Adjusted permissions and file-saving paths to improve file access support on Android tablets. Fixes #18076 and Fixes #19053 ([Commit](https://github.com/status-im/status-desktop/commit/0deb35448fda6f5d0195798b9ddee21650812dfc)).
> 
> The image save and upload needs a little bit of work
> 
> 1. Save doesn't work for me. Maybe we should use a simpler approach where `save` would just > save the file in Downloads
> 
> Video
> 
>  Screen_Recording_20251027_091456_Files.mp4 

Yup, separate code flow which I only discovered later; I can fix in a separate task/PR

> 2. Not sure if it's a nitpick or a must, but I would expect the image upload to open the android image selector or a fully fledged file selector. Similar to Discord or any other app. As opposed to the limited file selector with an image filter.
> 
> Discord vs Status - no idea how to find my images to send in Status
> 
> https://photos.app.goo.gl/E3Bi46DFYvH19ESG8

Yeah... I still haven't managed to persuade the file selector to go into an "image picker" mode :(

> 3. For some reason there's a regression on IOS alone. When selecting the wrong file format I'm not getting the error popup. It's just silence. Works fine on the latest TestFlight app.

OK, I'll look into that
